File: internal/content/testdata/bench/go/nc-valid-sudoku.go
func is_valid_sudoku(board [][]string) bool {
	rows := make([]map[string]bool, 9)
	cols := make([]map[string]bool, 9)
	boxes := make([]map[string]bool, 9)
	for i := 0; i < 9; i++ {
		rows[i] = make(map[string]bool)
		cols[i] = make(map[string]bool)
		boxes[i] = make(map[string]bool)
	}
	for r := 0; r < 9; r++ {
		for c := 0; c < 9; c++ {
			val := board[r][c]
			if val == "." {
				continue
			}
			boxIdx := (r/3)*3 + (c / 3)
			if rows[r][val] || cols[c][val] || boxes[boxIdx][val] {
				return false
			}
			rows[r][val] = true
			cols[c][val] = true
			boxes[boxIdx][val] = true
		}
	}
	return true
}

File: internal/content/testdata/bench/rust/nc-valid-sudoku.rs
use std::collections::HashSet;

fn is_valid_sudoku(board: Vec<Vec<String>>) -> bool {
    let mut rows: Vec<HashSet<String>> = (0..9).map(|_| HashSet::new()).collect();
    let mut cols: Vec<HashSet<String>> = (0..9).map(|_| HashSet::new()).collect();
    let mut boxes: Vec<HashSet<String>> = (0..9).map(|_| HashSet::new()).collect();
    for r in 0..9 {
        for c in 0..9 {
            let val = &board[r][c];
            if val == "." {
                continue;
            }
            let box_idx = (r / 3) * 3 + (c / 3);
            if rows[r].contains(val) || cols[c].contains(val) || boxes[box_idx].contains(val) {
                return false;
            }
            rows[r].insert(val.clone());
            cols[c].insert(val.clone());
            boxes[box_idx].insert(val.clone());
        }
    }
    true
}
